app/src/main/kotlin/lv/rigadevday/android/ui/splash/SplashActivity.kt

```diff
@@ -20,13 +31,28 @@ class SplashActivity : AppCompatActivity() {
 
         // Small warmup for Firebase and Rx caches
         Completable.concat(listOf(
-            repo.speakers().toList().toCompletable(),
-            repo.sessions().toList().toCompletable(),
-            repo.schedule().toList().toCompletable()
-        )).subscribe {
-            val intent = Intent(this, TabActivity::class.java)
-            startActivity(intent)
-            finish()
-        }
+                repo.speakers().toList().toCompletable(),
+                repo.sessions().toList().toCompletable(),
+                repo.schedule().toList().toCompletable()
+        ))
+                // If there is no internet or persistent data, call timeout
+                .timeout(TIME_OUT_SECONDS, TimeUnit.SECONDS, AndroidSchedulers.mainThread())
+                .subscribe(
+                        {
+                            val intent = Intent(this, TabActivity::class.java)
+                            startActivity(intent)
+                            finish()
+                        },
+                        {
+                            showMessage(R.string.error_connection_message)
+                            exitApp()
+                        }
+                )
+    }
+
+    fun exitApp() {
+        val handler = Handler()
```

Comment:
```
fun exitApp() = Handler().postDelayed(
   Runnable{ finish() },
   TIME_TO_EXIT
)
```

Looks better since it does not introduce throwaway variables and is less indented.